README.md

```diff
@@ -1,104 +1,394 @@
+[![License](https://img.shields.io/badge/License-Apache%202.0-green.svg)](LICENSE)
+[![GitHub stars](https://img.shields.io/github/stars/EvolvingLMMs-Lab/lmms-engine?style=social)](https://github.com/LMMs-Lab/lmms-engine/stargazers)
+
+[Quick Start](#-quick-start) • [Examples](#-featured-examples) • [Model Support](#-model-support) • [Optimizations](#️-optimizations) • [Architecture](#️-architecture) • [Documentation](#-documentation)
+
+</div>
+
+---
+
+## Overview
+
+**LMMs Engine** is a highly efficient, modular training framework for training Unified Multimodal Models at scale.
+
+Train any multimodal model architectures including vision-language models (Qwen2.5/3-VL, LLaVA-OV), diffusion models (dLLM, WanVideo), unified multimodal models (Qwen2.5-Omni, BAGEL) and specialized research architectures (RAE, Gated DeltaNet, SiT). 
+
+Built with distributed training optimizations (FSDP2 Multi-dimensional Parallelism, Ulysses Sequence Parallel, Flash Attention 2, Liger Kernel, Muon optimizer, Native Sparse Attention) and a modular design for easy extensibility.
+
+## 🚀 Quick Start
+
+### Installation
 
 ```bash
+# Clone the repository
+git clone https://github.com/LMMs-Lab/lmms-engine.git
+cd lmms-engine
+
+# Install dependencies
 uv sync
+
+# Optional: Performance optimizations
+uv pip install flash-attn --no-build-isolation
+uv pip install liger-kernel
 ```
 
-## Launch
+### Launch Training
 
-The recommended way to launch is always use torchrun as it is the most native way to launch torch and in most of the settings this should work. Most of the debug and development should be based on this as we might not always use accelerate in our later framework.
+**Recommended: torchrun (native PyTorch)**
+```bash
+torchrun --nproc_per_node=8 --nnodes=1 --node_rank=0 \
+  --master_addr=127.0.0.1 --master_port=12355 \
+  -m lmms_engine.launch.cli config_yaml=examples/qwen3_vl/example_config.yaml
+```
 
+**Alternative: Accelerate**
 ```bash
-torchrun --nproc_per_node=8 --nnodes=1 --node_rank=0 --master_addr=127.0.0.1 --master_port=12355 -m lmms_engine.launch.cli --config examples/load_from_pretrained_example.yaml
+accelerate launch --use_fsdp \
+  -m lmms_engine.launch.cli config_yaml=examples/qwen3_vl/example_config.yaml
 ```
 
-## Examples
+**Single GPU**
+```bash
+python -m lmms_engine.launch.cli config_yaml=examples/qwen3_vl/example_config.yaml
+```
 
-We provide two examples here to demonstrate how to use the training engine in most of the case, you will need to perform the following three steps:
+## 🔥 Featured Examples
 
-1. Process the dataset into a specific format and store it in (jsonl/json/arrow)
-2. Write your dataset yaml (Optional if you are only using a single data source)
-3. Prepare your training config
+| Model | Architecture | FSDP2 | Ulysses SP | Muon | Packing | NSA | Highlights | Quick Start |
```

Comment:
I think all models can be compatible with the muon optimizer, though they have not been fully tested yet.